source/rust_verify/src/rust_to_vir_func.rs

```diff
@@ -519,6 +700,91 @@ fn is_mut_ty<'tcx>(
     }
 }
 
+fn predicates_match<'tcx>(
+    tcx: TyCtxt<'tcx>,
+    id1: rustc_span::def_id::DefId,
+    id2: rustc_span::def_id::DefId,
+    substs: SubstsRef<'tcx>,
+) -> bool {
+    let preds1 = all_predicates(tcx, id1, substs);
+    let preds2 = all_predicates(tcx, id2, substs);
+    if preds1.len() != preds2.len() {
+        return false;
+    }
+    // TODO should allow this to work if the predicates are in a different order
+    for (p1, p2) in preds1.iter().zip(preds2.iter()) {
+        let p1 = tcx.anonymize_bound_vars(p1.kind());
+        let p2 = tcx.anonymize_bound_vars(p2.kind());
+        if p1 != p2 {
+            return false;
+        }
+    }
+    return true;
+}
+
+fn all_predicates<'tcx>(
+    tcx: TyCtxt<'tcx>,
+    id1: rustc_span::def_id::DefId,
+    substs: SubstsRef<'tcx>,
+) -> Vec<Predicate<'tcx>> {
+    let preds = tcx.predicates_of(id1);
+    // TODO need to fix this when external_fn_specification handles associated methods / member functions
+    assert!(preds.parent.is_none());
+    let preds = preds.instantiate(tcx, substs);
+    preds.predicates
+}
+
+pub(crate) fn get_external_def_id<'tcx>(
+    tcx: TyCtxt<'tcx>,
+    body_id: &BodyId,
+    body: &Body<'tcx>,
+    sig: &'tcx FnSig<'tcx>,
+) -> Result<rustc_span::def_id::DefId, VirErr> {
+    let err = || {
+        err_span(
+            sig.span,
+            format!("external_fn_specification encoding error: body should end in call expression"),
+        )
+    };
+
+    // Get the 'body' of this function (skipping over header if necessary)
+    let expr = match &body.value.kind {
+        ExprKind::Block(block_body, _) => match &block_body.expr {
+            Some(body_value) => body_value,
+            None => {
+                return err();
+            }
+        },
+        _ => &body.value,
+    };
+
+    // TODO handle MethodCall
```

Comment:
Should this TODO be resolved? Or is it intentionally left for later?
I believe `MethodCall` is only relevant for associated methods (or trait impls), so I think that branch is currently guarded by the "unsupported" error in `rust_to_vir_func`. If this is intentionally left for later; can you expand on the comment, to point out that `MethodCall` would be unreachable currently?

Reply:
it's left for later, just a matter of scoping down the PR.